lib/fedora_migrate/datastream_verification.rb

```diff
@@ -0,0 +1,43 @@
+module FedoraMigrate::DatastreamVerification
+  
+  attr_accessor :datastream
+
+  def valid? datastream=nil
+    @datastream = datastream || source
+    return true if no_checksum?
+    return true if checksums_match?
+    return true if content_equivalent? 
+    FedoraMigrate::Logger.warn "#{source.pid} datastream #{datastream.dsid} validation failed"
+    return false
+  end
+
+  def checksums_match?
+    datastream.checksum == target_checksum.split(/:/).last
+  end
+
+  # TODO: Reporting mechanism? If there isn't a checksum it defaults to "none" (issue #4)
+  def no_checksum?
+    if datastream.checksum == "none"
+      FedoraMigrate::Logger.info "unable to vaidate datastream because the checksum was 'none'"
+      return true
+    end
+  end
+  
+  def content_equivalent?
+    if datastream.mimeType == "text/xml"
+      EquivalentXml.equivalent?(target_content, datastream.content)
```

Comment:
The checksums of xml datastreams aren't matching because the line `<?xml version="1.0"?>` is getting added. Why that's happening, who knows. I'm seeing Fedora4 do it. We should delve into that, but for now we ought to have this work-around to ensure the content is correct.